tests/test_io.py

```diff
@@ -475,20 +477,29 @@ def test_gwy_read_component(load_scan_dummy: LoadScans) -> None:
         assert list(test_dict.values()) == [{"test nested component": 3}]
 
 
-# FIXME : Get this test working
-# @pytest.mark.parametrize(
-#     "unit, x, y, expected",
-#     [
-#         ("um", 100, 100, 97.65625),
-#         ("nm", 50, 50, 0.048828125),
-#     ],
-# )
-# def test_extract_pixel_to_nm_scaling(load_scan: LoadScans, unit, x, y, expected) -> None:
-#     """Test extraction of pixels to nanometer scaling."""
-#     load_scan.load_spm()
-#     load_scan._spm_pixel_to_nm_scaling() {"unit": unit, "x": x, "y": y}
-#     test_filters_random.extract_pixel_to_nm_scaling()
-#     assert test_filters_random.pixel_to_nm_scaling == expected
+@patch("pySPM.SPM.SPM_image.pxs")
+@pytest.mark.parametrize(
+    ("unit", "x", "y", "expected_px2nm"),
+    [
+        ("mm", 0.01, 0.01, 10000),
+        ("um", 1.5, 1.5, 1500),
+        ("nm", 50, 50, 50),
+        ("pm", 233, 233, 0.233),
```

Comment:
Made some suggestions in #801 to test the scenario that cause this error, not sure if they make sense but when bugs occurs its useful to test, at least in some manner, that the scenario that caused them is handled correctly.

Reply:
comment left on #801, thanks for the extra tests